lib/csv/table.rb

```diff
@@ -23,8 +23,19 @@ class Table
     # * length()
     # * size()
     #
-    def initialize(array_of_rows)
-      @table = array_of_rows
+    def initialize(array_of_rows, headers=nil)
+      if headers.nil?
+        @table = array_of_rows
+        if @table.empty?
+          @headers = []
+        else
+          @headers = @table.first.headers
+        end
+      else
+        @headers = headers
+        @table = array_of_rows
```

Comment:
There's a duplication of code here: the `@table` assignment. We should keep only one at the beginning of `initialize`.

Reply:
I've moved the @table assignment to the beginning of initialize.